server/monitor.go

```diff
@@ -58,6 +60,8 @@ type Monitor struct {
 	callback          MonitorCallback
 }
 
+var ErrSetNotFound = errors.New("set not found in monitor heap")
```

Comment:
Should not use internal implementation details in user error message. "set not monitored by server"?
Also, "ErrSetNotFound" is too generic for a monitor-heap-specific error. ErrSetNotMonitored?